### PR TITLE
scripts: install gnupg only when apt-key is needed

### PR DESCRIPTION
**Pull request by simondeziel**
apt-key depends on gnupg but apt-key itself if not used
on modern systems (APT_KEY_TYPE=keyring).

Signed-off-by: Simon Deziel <simon@sdeziel.info>

### CODE REVIEW

**Comment by bradfitz** [comment]
I'll defer to @danderson or @Xe here.

File: scripts/installer.sh
Comment (simondeziel) [reply scripts/installer.sh:410]: In the APT_KEY_TYPE == "keyring" case, no conversion is needed as `apt` itself will understand the binary keyring saved in /usr/share/keyrings/tailscale-archive-keyring.gpg.

In the APT_KEY_TYPE == "legacy" case, `apt-key` is used to take the `.asc` key and turn it into a binary keyring living in /etc/apt/trusted.gpg. To do the `.asc` to `.gpg` conversion, it relies on gnupg.

Comment (danderson) [reply scripts/installer.sh:410]: Yup, this looks right to me. Thanks!

**Review by danderson — APPROVED** [review]

**Comment by danderson** [comment]
Letting CI run before merging (it was blocked waiting on maintainer approval), but will merge shortly. Thanks!

**Comment by bradfitz** [comment]
> Letting CI run before merging (it was blocked waiting on maintainer approval), but will merge shortly. Thanks!

The "Approve and run" button seems broken right now. It does nothing.

So I'm just merging. This file has no CI coverage anyway.

**Pull request merged**
